### PR TITLE
Remove sharding code

### PR DESCRIPTION
**Pull request by jannikluhn**
### What was wrong?

#508 

### How was it fixed?

Deleted most of the sharding code. Only collation and collation header remain. I tried (semi-successfully) to split it up into small commits so that reviving becomes easier.

This should be merged only after the SMC and SMC handler have been transferred to the sharding repository.

#### Cute Animal Picture

put a cute animal picture link inside the parentheses

### CODE REVIEW

File: evm/exceptions.py [hunk 1]
```diff
@@ -151,14 +151,6 @@ class OutOfBoundsRead(VMError):
...
 
-class UnannouncedStateAccess(VMError):
-    """
-    Error raised if a storage slot or account field is accessed that has not
-    been specified in the transaction's read or write list, respectively.
-    """
-    pass
-
-
 class GasPriceAlreadySet(Exception):
```
Comment (pipermerriam) [inline evm/exceptions.py]: Should this get remove since I believe it is `PAYGAS` specific.

**Comment by hwwhww** [comment]
And remove `eth-tester` from setup.py?

File: evm/exceptions.py (hunk 1 above)
Comment (jannikluhn) [reply evm/exceptions.py]: :+1: missed this (same for `NotTopLevelCall`)

**Comment by hwwhww** [comment]
Also, remove `TrackedDB(BaseDB)` which was for the stateless client? And it requires removing the all `access_logs` stuff.

File: tests/core/helpers.py [hunk 2]
```diff
@@ -1,7 +1,3 @@
-from eth_utils import decode_hex
-
-from evm.utils.padding import pad32
-from evm.vm.forks.sharding.transactions import ShardingTransaction
```
Comment (hwwhww) [inline tests/core/helpers.py]: Remove two empty lines.

File: setup.py [hunk 3]
```diff
@@ -26,7 +26,6 @@
         "rlp>=0.4.7,<1.0.0",
         "eth-keys>=0.2.0b3,<1.0.0",
         "trie>=1.3.0,<2.0.0",
-        "eth-tester==0.1.0b21",
         "web3>=4.0.0b6,<5.0.0",
```
Comment (hwwhww) [inline setup.py]: oh sorry, also remove `web3`?

File: tests/core/helpers.py (hunk 2 above)
Comment (jannikluhn) [reply tests/core/helpers.py]: Flake8 fails on my machine if I do this (`tests/core/helpers.py:1:1: E302 expected 2 blank lines, found 0`). Pushed this anyway, let's see what Travis thinks about this.

Comment (jannikluhn) [reply tests/core/helpers.py]: Apparently it's a known bug in pep8/pycodestyle: https://github.com/PyCQA/pycodestyle/issues/473

I've added a noqa for now.

**Comment by jannikluhn** [comment]
Rebased and removed recently added sharding code (shard tracker and some SMC stuff). Will merge when the tests have passed.

**Comment by carver** [comment]
> I had to add eth-hash as a dependency

eth-hash is already loaded by eth-utils, but web3.py loads a specific eth-hash backend (pycryptodome). In keeping with the eventual idea of py-evm becoming a library/utility, and trinity being the full-featured node, py-evm should probably _not_ be opinionated about the hashing backend used.

Instead, we can add a backend only during testing, like how eth-utils does it:
https://github.com/ethereum/eth-utils/blob/master/tox.ini#L14-L15

**Comment by jannikluhn** [comment]
Good point, I've added it to `dev-requirements.txt`. Can I merge this now?

**Comment by NIC619** [comment]
There seems to be `access_list` left uncleaned in `Message`(`vm/message.py`) and `state_db()`(`vm/state.py`)?

**Pull request merged**